### PR TITLE
Check that generated compat-data is up to date on CI

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
| Q                        | A 
| ------------------------ | ---
| Tests Added + Pass?      | Yes
| License                  | MIT

This will prevent cases like 6363b594cc96caa55b293bad4da1d948a31d1702: the file used to generate the data was updated but the data itself wasn't, so we didn't notice the regression in the PR which updated the compat-table mappings.

### CODE REVIEW

File: packages/babel-compat-data/scripts/build-data.js [hunk 1]
```diff
@@ -297,19 +296,19 @@ const generateData = (environments, features) => {
...
 
-    if (!isEqual(currentData, newData)) {
+    // Compare as JSON strings to also check keys ordering
+    if (currentData !== stringified) {
       console.error(
         "The newly generated plugin/built-in data does not match the current " +
-          "files. Re-run `npm run build-data`."
+          "files. Re-run `make build-compat-data`."
       );
       process.exit(1);
```
Comment (JLHwung) [inline packages/babel-compat-data/scripts/build-data.js]:
```suggestion
      process.exitCode = 1;
```

Comment (nicolo-ribaudo) [reply packages/babel-compat-data/scripts/build-data.js]: Why?

Comment (JLHwung) [reply packages/babel-compat-data/scripts/build-data.js]: > In most situations, it is not actually necessary to call process.exit() explicitly.

https://nodejs.org/api/process.html#process_process_exit_code

File: packages/babel-compat-data/scripts/build-data.js [hunk 2]
```diff
@@ -306,7 +306,7 @@ const generateData = (environments, features) => {
         "The newly generated plugin/built-in data does not match the current " +
           "files. Re-run `make build-compat-data`."
       );
-      process.exit(1);
```
Comment (JLHwung) [inline packages/babel-compat-data/scripts/build-data.js]: Can you revert this commit? I realize that this function is in a `forEach` callback.

**Pull request merged**